autoload/db_ui/dbout.vim

```diff
@@ -123,22 +126,46 @@ function! db_ui#dbout#yank_header() abort
   call setreg(v:register, csv_columns)
 endfunction
 
-function! s:get_cell_range(line, col) abort
+function! s:get_cell_range(cell_line_number, curpos, scheme_name) abort
+  let line = getline(a:cell_line_number)
+  let position = a:curpos[1:]
   let table_line = '-'
-  let col = a:col - 1
-  let from = 0
-  let to = 0
-  while col >= 0 && a:line[col] ==? table_line
-    let from = col
+
+  let col = position[-1] - 1
+  let virtual_from = 0
+
+  while col >= 0 && line[col] ==? table_line
+    let virtual_from = col
     let col -= 1
   endwhile
-  let col = a:col - 1
-  while col <= len(a:line) && a:line[col] ==? table_line
-    let to = col
+
+  let col = position[-1] - 1
+  let virtual_to = 0
+
+  while col <= len(line) && line[col] ==? table_line
+    let virtual_to = col
     let col += 1
   endwhile
 
-  return {'from': from, 'to': to}
+  let position_above = insert(position[1:], position[0] - 1)
+
+  call cursor(add(position_above[:-2], virtual_from))
+  norm! j
+  let from = col('.')
+
+  call cursor(add(position_above[:-2], virtual_to))
+  norm! j
+  let to = col('.')
+
+  call cursor(position)
+
+  " NOTE: 'virtual' refers to a position in reference to virtcol().
+  "       other fields reference col()
+  return {
```

Comment:
Now, we return a dictionary that has a different structure. This dictionary maintains the original locations of the "virtual" column space (aka, what the column _should_ be when there are no tabs) as well as the "actual" locations of each column (aka, what the column _actually_ is when there are tabs in the line).